pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -4925,3 +4925,74 @@ status:
 		})
 	}
 }
+
+func TestReconcile_ReplacementsInStatusTaskSpec(t *testing.T) {
+	task := parse.MustParseTask(t, `
+metadata:
+  name: test-task-with-replacements
+  namespace: foo
+spec:
+  params:
+  - default: mydefault
+    name: myarg
+    type: string
+  steps:
+  - script: echo $(inputs.params.myarg)
+    image: myimage
+    name: mycontainer
+`)
+	tr := parse.MustParseTaskRun(t, `
+metadata:
+  name: test-taskrun-with-replacements
+  namespace: foo
+spec:
+  params:
+  - name: myarg
+    value: foo
+  taskRef:
+    name: test-task-with-replacements
+status:
```

Comment:
I initially tried this with a pre-set `tr.Status.TaskSpec`, but that kept passing when it shouldn't have. I'm guessing it might have eventually flaked into failure, but it's very possible that the specific conditions that triggered the flaky behavior would never actually show up in a unit test, so I just went with testing to make sure that a `nil` `tr.Status.TaskSpec` gets written properly.